docs/developer/languages/go/components.mdx

```diff
@@ -0,0 +1,413 @@
+
+[`wadge`](https://github.com/wasmCloud/wadge) is a "bridging" framework that enables your native code to interact with WebAssembly component interfaces for purposes such as testing. `wadge` acts as a bridge between your Go toolchain and a WebAssembly runtime that makes interfaces “just work” when it's time to test your code. 
+
+## Get started
+
+In this walkthrough, we will create an HTTP server from scratch using the [Go component SDK](https://github.com/wasmCloud/component-sdk-go/), write a test for the application, and use `wadge` to run the test. This walkthrough requires:
+
+* [Go](https://go.dev/doc/install) 1.23.0+
+* [TinyGo](https://tinygo.org/getting-started/install/) 0.33.0+
+* [wasmCloud Shell (`wash`)](https://wasmcloud.com/docs/installation) CLI 0.32.1+ for building and deploying components
+* [`wit-deps`](https://github.com/bytecodealliance/wit-deps) 0.3.4+ for WIT dependency management
+
+### Step 1: Set up with `wash`
+
+The wasmCloud Shell (`wash`) CLI helps developers build component-based applications that can be deployed with wasmCloud, and consolidates many open source component development tools. 
+
+Create a new project called `http-test` and navigate into the project directory:
+
+```shell
+wash new component http-test --template-name hello-world-tinygo
+```
+```shell
+cd http-test
+```
+
+Replace the contents of `wit/deps.toml` with the WIT dependency targets below:
+
+```toml
+http = "https://github.com/WebAssembly/wasi-http/archive/v0.2.0.tar.gz"
+component = "https://github.com/wasmCloud/component-sdk-go/archive/main.tar.gz"
+```
+Run `wit-deps` from the root of your project directory:
+
+```shell
+wit-deps
+```
+
+Replace the contents of `world.wit` with the WIT definition below:
+
+```wit
+package example:http-server;
+
+world hello {
+  include wasmcloud:component/imports;
+  export wasi:http/incoming-handler@0.2.0;
+}
+```
+
+Finally, we'll delete the contents of `hello.go` and write our HTTP server, which will return a simple "hello world."
+
+Using the Component SDK, this looks like a fairly standard server using the HTTP standard library, with only a handful of exceptions where we use methods of `wasihttp` or `wasilog`.
+
+```go
+//go:generate go run github.com/bytecodealliance/wasm-tools-go/cmd/wit-bindgen-go generate --world example --out gen ./wit
+
+package main
+
+import (
+	"net/http"
+
+	"go.wasmcloud.dev/component/log/wasilog"
+	"go.wasmcloud.dev/component/net/wasihttp"
+)
+
+func init() {
+	wasihttp.HandleFunc(handler)
+}
+
+func handler(w http.ResponseWriter, r *http.Request) {
+	logger := wasilog.ContextLogger("handler")
+
+	logger.Info("request received", "host", r.Host, "path", r.URL.Path, "agent", r.Header.Get("User-Agent"))
+
+	_, err := w.Write([]byte("hello world!"))
+	if err != nil {
+		logger.Error("failed to write body", "error", err)
+	}
+}
+
+func main() {}
+```
+
+Download missing packages:
+
+```shell
+go mod tidy
+```
+Run `wash build` to generate bindings:
+
+```shell
+wash build
```

Comment:
note: this failed for me, perhaps we can note _for now_ that this will generate an error but that's okay. We'll of course make this better by just building the build command into the wasmcloud.toml itself

Reply:
BTW here's the wasmcloud.toml that made this work

```toml
name = "http-hello-world"
language = "tinygo"
type = "component"
version = "0.1.0"

[component]
wasm_target = "wasm32-wasi"
destination = "build/http_hello_world_s.wasm"
build_command = "tinygo build --target=wasip2 --wit-package ./wit --wit-world hello"
build_artifact = "http-test.wasm"
```

Reply:
Huh, do you get an error on build? It works on my end. I'll add the you-may-get-an-error-but-that's-okay note in the meantime.

Reply:
I do, but weird 🤔 Maybe it was my issue. Lemme run through with the new instructions